scripts/datasets/process_nerfstudio_to_sdfstudio.py

```diff
@@ -12,255 +11,258 @@
-            cy.append(camera["cy"])
-            fl_x.append(camera["fl_x"])
-            fl_y.append(camera["fl_y"])
-
+    for frame in frames:
+        # load intrinsics from polycam
+        if args.data_type == "polycam":
+            cam_intrinsics.append(np.array([
+                [frame["fl_x"], 0, frame["cx"]],
+                [0, frame["fl_y"], frame["cy"]],
+                [0, 0, 1]]))
+
+        # load poses
         # OpenGL/Blender convention, needs to change to COLMAP/OpenCV convention
         # https://docs.nerf.studio/en/latest/quickstart/data_conventions.html
-        ## IGNORED for now
-        c2w = np.array(camera["transform_matrix"]).reshape(4, 4)
+        # IGNORED for now
+        c2w = np.array(frame["transform_matrix"]).reshape(4, 4)
         c2w[0:3, 1:3] *= -1
-
         poses.append(c2w)
 
-        file_path = Path(camera["file_path"])
-        img_path = input_path / "images" / file_path.name
+        # load images
+        file_path = Path(frame["file_path"])
+        img_path = input_dir / "images" / file_path.name
         assert img_path.exists()
         image_paths.append(img_path)
 
-        # include sensor depths
-        if args.geo_type == "sensor_depth":
-            depth_path = input_path / "depths" / f"{file_path.stem}.png"
-            assert depth_path.exists(), f"depth path {depth_path} does not exist. Only available for polycam data."
+        # load sensor depths
+        if args.data_type == "polycam" and args.sensor_depth:
+            depth_path = input_dir / "depths" / f"{file_path.stem}.png"
+            assert depth_path.exists()
             depth_paths.append(depth_path)
 
-    poses = np.array(poses)
+    # Check correctness
+    assert len(poses) == len(image_paths)
+    assert len(poses) == len(cam_intrinsics) or len(cam_intrinsics) == 1
 
-    if POLYCAM:
-        # intrinsics
-        camera_intrinsics = []
-        for idx in range(len(cx)):
-            camera_intrinsics.append(np.array([[fl_x[idx], 0, cx[idx]], [0, fl_y[idx], cy[idx]], [0, 0, 1]]))
-    else:
-        camera_intrinsics = np.array([[fl_x, 0, cx], [0, fl_y, cy], [0, 0, 1]])
-
-    # deal with invalid poses
+    # Filter invalid poses
+    poses = np.array(poses)
     valid_poses = np.isfinite(poses).all(axis=2).all(axis=1)
     min_vertices = poses[:, :3, 3][valid_poses].min(axis=0)
     max_vertices = poses[:, :3, 3][valid_poses].max(axis=0)
 
-    # camera pose normalization only used for indoor scenes
-    if args.indoor:
-        center = (min_vertices + max_vertices) / 2.0
-        scale = 2.0 / (np.max(max_vertices - min_vertices) + 3.0)
-
-        # we should normalize pose to unit cube
-        poses[:, :3, 3] -= center
-        poses[:, :3, 3] *= scale
-
-        # inverse normalization
+    # === Normalize the scene ===
+    if args.scene_type in ["indoor", "object"]:
+        scene_center = (min_vertices + max_vertices) / 2.0
+        scene_scale = 2.0 / (np.max(max_vertices - min_vertices) * 1.05)
```

Comment:
I think we should normalize indoor scenes and object scenes differently. If we use this scene scale in door scenes, most of part of the scenes (e.g. walls) will be outside of the [-1, 1] cube.

Reply:
I see. This part was not changed compared to the original code. (Actually, I was confused about this indoor normalization step either.)

Is there an elegant method to normalize indoor scenes if don't have the depth or point cloud prior? Set it with a "safe" value?

Reply:
Yes, I agree the normalisation for indoor scenes is not elegant. It's mainly based on ScanNet dataset where the cameras are looking outwards, the camera poses are in meter space and are roughly 1.5 meters to the wall. Maybe we could support a far value for each camera and then compute the scale with `(max - min) + 2 * far`. But we need to set the `far` in the same metric space as camera poses which is not easy to choose if the camera poses are computed with colmap.

Reply:
Great description, thanks!

Additionally, I think we are still likely not to determine the `far` easily even if unify the metric space into meter space because the area of the indoor scene may vary from each other. (For instance, factory v.s. home.) Maybe the depth prior is necessary.

Could we use `np.max(max_vertices - min_vertices) * 5` temporarily? Will this setting worsen the small indoor scenes much?

Reply:
Added an arg `--scene-scale-mult`. Set 1.05 / 5.0 for object / indoor when the user doesn't explicitly set it.